### DISCUSSION TITLE
Best model and best parameter check krna ka way

### DISCUSSION THREAD

**Discussion by Saibtain** [AammarTufail/python_ka_chilla2023#89] [Q&A]
kia hm sara model ko aik sath rakh ka or grid search cv ko import kr ka or sb model mean 3,4 model ka paraim_ grid aik sath da kr phir jo model best ho sirf osi ki best score show ho with parameter

**Comment by usamabajwa86** [comment]
yes ho sakta hai

**Comment by Saibtain** [comment]
But execute nhi ho rahy kb ki try kr rahy hn

**Reply by Saibtain** [reply]
from sklearn.model_selection import GridSearchCV
sorted_models = grid_search
param_grid = [{'n_neighbors': np.arange(1,30,2) , 'weights': ['uniform', 'distance']},
               {'max_depth': [3,5,7, None] , 'min_samples_split': [2,3,4]},
               {'n_neighbors': [1,3,6,9], 'weights': ['uniform', 'distance']} ] 
grid_search = GridSearchCV(model, param_grid, scoring='precision')

grid_search = GridSearchCV(estimator=[DecisionTreeClassifier(),KNeighborsClassifier(),
              RandomForestClassifier() ])
grid_search.fit(X,y)
# print the best parameters
print("Best Parameters: ", grid_search.best_params_)
print("Best Score: ", grid_search.best_score_)



oper wala model wohi ha necha ma yee try kr rahy hn tak connection bn saka

**Comment by Saibtain** [comment]
GridSearchCV.__init__() missing 1 required positional argument: 'param_grid
yee error aa rahy ha

**Comment by mhamza6051** [comment]
i have also same error
![11](https://user-images.githubusercontent.com/121826823/214328769-a36f7de2-4b5a-4653-b295-276465fe548b.jpg)
![22](https://user-images.githubusercontent.com/121826823/214328797-0bf5ffc6-60bb-43d0-95a1-db3a5d1bb921.jpg)

**Comment by Saibtain** [comment]
bahi hm sb aik group ma han but problem solve nhi kr saka
kia  hm aik group seprat nhi bna lata jis ma hm aik dosra sa kisi bhi trah
ka question  discus  sakany  or phir jb sb sa na ho tb sir sa meeting ma
discuss krany

daily base pa 30 mint jb ap sb practice krna lagta han tb aik dosra sa zoom
meeting pa discuss kr lia krany

**Reply by mhamza6051** [reply]
best ha

**Comment by shahbaz-prog** [comment]
<img width="516" alt="image" src="https://user-images.githubusercontent.com/119630744/214455657-51c8e4e7-bccd-4faf-9ad7-7ae6e9ddc0eb.png">

**Comment by shahbaz-prog** [comment]
<img width="473" alt="image" src="https://user-images.githubusercontent.com/119630744/214455782-f1aadd0d-6639-4f34-b6e5-e82f5637b52e.png">

**Reply by shahbaz-prog** [reply]
This above code is running fine if we take out KneighborsClassifier from the iteration ??
